libtock/usb_keyboard_hid.c

```diff
@@ -0,0 +1,232 @@
+    case '%':   *modifier = shift;
+      *key = 0x22;
+      return 0;
+    case '^':   *modifier = shift;
+      *key = 0x23;
+      return 0;
+    case '&':   *modifier = shift;
+      *key = 0x24;
+      return 0;
+    case '*':   *modifier = shift;
+      *key = 0x25;
+      return 0;
+    case '(':   *modifier = shift;
+      *key = 0x26;
+      return 0;
+    case ')':   *modifier = shift;
+      *key = 0x27;
+      return 0;
+    case '0':   *key = 0x27;
+      return 0;
+    case '\n':  *key = 0x28;
+      return 0;                          // enter
+    case '\r':  *key = 0x28;
+      return 0;                          // enter
+    case '\b':  *key = 0x2A;
+      return 0;                          // backspace
+    case '\t':  *key = 0x2B;
+      return 0;                          // tab
+    case ' ':   *key = 0x2C;
+      return 0;                          // space
+    case '_':   *modifier = shift;
+      *key = 0x2D;
+      return 0;
+    case '-':   *key = 0x2D;
+      return 0;
+    case '+':   *modifier = shift;
+      *key = 0x2E;
+      return 0;
+    case '=':   *key = 0x2E;
+      return 0;
+    case '{':   *modifier = shift;
+      *key = 0x2F;
+      return 0;
+    case '[':   *key = 0x2F;
+      return 0;
+    case '}':   *modifier = shift;
+      *key = 0x30;
+      return 0;
+    case ']':   *key = 0x30;
+      return 0;
+    case '|':   *modifier = shift;
+      *key = 0x31;
+      return 0;
+    case '\\':   *key = 0x31;
+      return 0;
+    case ':':   *modifier = shift;
+      *key = 0x33;
+      return 0;
+    case ';':   *key = 0x33;
+      return 0;
+    case '"':   *modifier = shift;
+      *key = 0x34;
+      return 0;
+    case '\'':   *key = 0x34;
+      return 0;
+    case '~':   *modifier = shift;
+      *key = 0x35;
+      return 0;
+    case '`':   *key = 0x35;
+      return 0;
+    case '<':   *modifier = shift;
+      *key = 0x36;
+      return 0;
+    case ',':   *key = 0x36;
+      return 0;
+    case '>':   *modifier = shift;
+      *key = 0x37;
+      return 0;
+    case '.':   *key = 0x37;
+      return 0;
+    case '?':   *modifier = shift;
+      *key = 0x38;
+      return 0;
+    case '/':   *key = 0x38;
+      return 0;
+  }
+  return -1;
+}
+
+
+
+int usb_keyboard_hid_send_letter_sync(uint8_t* buffer, char letter) {
+  int err;
+
+  uint8_t modifier;
+  uint8_t key = 0;
+
+  to_hid_keycode(letter, &modifier, &key);
+
+  buffer[0] = modifier;
+  buffer[1] = 0;
+  buffer[2] = key;
+  buffer[3] = 0;
+  buffer[4] = 0;
+  buffer[5] = 0;
+  buffer[6] = 0;
+  buffer[7] = 0;
+
+  // Send key press.
+  err = usb_keyboard_hid_send_sync(buffer, 64);
+  if (err < 0) return err;
+
+  buffer[0] = 0;
+  buffer[2] = 0;
+
+  // Clear key press.
+  err = usb_keyboard_hid_send_sync(buffer, 64);
+  if (err < 0) return err;
+
+  return RETURNCODE_SUCCESS;
+
+}
+
+int usb_keyboard_hid_send_string_sync(uint8_t* buffer, char* str, int length) {
+  int i = 0;
+  int err;
+
+  for (i = 0; i < length; i++) {
```

Comment:
We don't check that length is smaller or equal to `64`

Reply:
It doesn't have to be less than 64. Only `buffer` needs to be 64 bytes.